include/CL/opencl.hpp

```diff
@@ -2432,7 +2437,7 @@ class Device : public detail::Wrapper<cl_device_id>
         static PFN_clCreateSubDevicesEXT pfn_clCreateSubDevicesEXT = nullptr;
 #if CL_HPP_TARGET_OPENCL_VERSION >= 120
         cl::Device device(object_);
-        cl_platform_id platform = device.getInfo<CL_DEVICE_PLATFORM>();
+        cl_platform_id platform = this->getPlatformIdFromDevice(device);
```

Comment:
I think this will work but the helper function seems like overkill, especially since we're just calling `operator()` on line 10468 (see below).  Why don't we just call `operator()` here also?
```suggestion
        cl_platform_id platform = device.getInfo<CL_DEVICE_PLATFORM>()();
```

Reply:
@bashbaug Do you have any other proposal?
That one which You proposed unfortunately seems is not compiling.
My one is compiling and working but I don't know why one of the CI jobs ran into timeout 
![image](https://user-images.githubusercontent.com/93128086/226652380-ba52aa5b-5d91-454d-9b01-ba63be3d69be.png)

I think that the best other solution would be to put every derived class in separate  header - but it is quite big change.
The easiest way in my opinion is to use my change.
What do You think?
Thanks!

Reply:
My other proposal is to move definition of `cl_int Device::createSubDevices(const cl_device_partition_property_ext* properties,
                        vector<Device>* devices)`
after the Platform class definition
This will also solve the issue.

Reply:
I just putted this new proposal about removing definition of `createSubDevices` after Platform class